content/_code-samples/airgapped-wallet/py/airgapped-wallet-TEST.py

```diff
@@ -0,0 +1,225 @@
+from xrpl.transaction import safe_sign_transaction
+
+
+def create_wallet(silent: False):
+    """
+    Generates a keypair
+    """
+    if not silent:
+        print("1. Generating seed...")
+        seed = keypairs.generate_seed()
+
+        print("2. Deriving keypair from seed...")
+        pub, priv = keypairs.derive_keypair(seed)
+
+        print("3. Deriving classic addresses from keypair..\n")
+        address = keypairs.derive_classic_address(pub)
+
+    else:
+        seed = keypairs.generate_seed()
+        pub, priv = keypairs.derive_keypair(seed)
+        address = keypairs.derive_classic_address(pub)
+ 
+    return address, seed
+
+
+def sign_transaction(xrp_amount, destination, ledger_seq, wallet_seq, password):
+    """
+    Signs transaction and returns signed transaction blob in QR code
+    """
+    print("1. Retrieving encrypted private key and salt...")
+    with open(get_path("/WalletTEST/private.txt"), "r") as f:
+        seed = f.read()
+        seed = bytes.fromhex(seed)
+
+    with open(get_path("/WalletTEST/salt.txt"), "rb") as f:
+        salt = f.read()
+
+    print("2. Initializing key...")
+    kdf = PBKDF2HMAC(
+        algorithm=hashes.SHA256(),
+        length=32,
+        iterations=100000,
+        salt=salt
+    )
+
+    key = base64.urlsafe_b64encode(kdf.derive(bytes(password.encode())))
+    crypt = Fernet(key)
+
+    print("3. Decrypting wallet's private key using password")
+    seed = crypt.decrypt(seed)
+
+    print("4. Initializing wallet using decrypted private key")
+    _wallet = wallet.Wallet(seed=seed.decode(), sequence=0)
+
+    validated_seq = ledger_seq
+    _wallet.sequence = wallet_seq
+
+    print("5. Constructing payment transaction...")
+    my_tx_payment = Payment(
+        account=_wallet.classic_address,
+        amount=xrp_to_drops(xrp=xrp_amount),
+        destination=destination,
+        last_ledger_sequence=validated_seq + 100,
+        # +100 to catch up with the ledger when we transmit the signed tx blob to Machine 2
+        sequence=_wallet.sequence,
+        fee="10"
+    )
+
+    print("6. Signing transaction...")
+    my_tx_payment_signed = safe_sign_transaction(transaction=my_tx_payment, wallet=_wallet)
+
+    img = qrcode.make(my_tx_payment_signed.to_dict())
+
+    print("7. Displaying signed transaction blob's QR code on the screen...")
+    img.save(get_path("/WalletTEST/transactionID.png"))
+    image = Image.open(get_path("/WalletTEST/transactionID.png"))
+    image.show()
+
+    print(f"RESULT: {my_tx_payment_signed.to_dict()}")
+    print("END RESULT: Successful")
+
+
+def get_path(file):
+    """
+    Get path (filesystem management)
+    """
+
+    global File_
+    # Checks what OS is being used
+    OS = platform.system()
+    usr = Path.home()
+
+    # Get PATH format based on the OS
+    if OS == "Windows":
+        File_ = PureWindowsPath(str(usr) + file)
+    if OS == "Linux":
+        File_ = PurePath(str(usr) + file)
```

Comment:
```suggestion
    # Get PATH format based on the OS
    if OS == "Windows":
        File_ = PureWindowsPath(str(usr) + file)
    else: # Assuming Linux-style file format
        File_ = PurePath(str(usr) + file)
```

My macbook is running `Darwin` so couldn't run the code. I think using else is good enough usually.

Reply:
(This has ripple effects down below since there are other checks for Linux which would need to be updated as well)

Reply:
I see, thank you.

Reply:
There are still 3 places to make this change - One in `airgapped-wallet-TEST.py` and two in `airgapped-wallet.py` (Searching for 'OS == "Linux"' should reveal them all)